props/saber_fett263_buttons.h

```diff
@@ -293,248 +659,3857 @@ class SaberFett263Buttons : public virtual PropBase {
+         MenuSave();
+         break;
+      case MENU_RETRACTION_OPTION:
+      case MENU_RETRACTION_TIME:
+         menu_type_ = MENU_STYLE_SETTING_SUB;
+         char style_arg[10];
+         itoa(ignite_time_, style_arg, 10);
+         current_preset_.SetStyle(blade_num_,style_parser.SetArgument(current_preset_.GetStyle(blade_num_), 19, style_arg));
+         current_preset_.Save();
+         MenuSave();
+         break;
+      case MENU_CLASH_LOCATION:
+      case MENU_DRAG_SIZE:
+      case MENU_EMITTER_SIZE:
+        SaveStyleSetting();
+        break;
+      case MENU_DELETE:
+        if (choice_) {
+          current_preset_.SaveAt(-1);
+          choice_ = false;
+          menu_type_ = MENU_TOP;
+          MenuSelect();
+          break;
+        }
+        PlayMenuSound("maffirm.wav");
+        break;
+      case MENU_EXIT:
+        PlayMenuSound("mselect.wav");
+        MenuExit();
+        break;
+  #endif
+      default:
+        break;  
+      }
+  }
+
+// Edit Mode Dial Right      
+  void MenuDialRight() {
+    switch (menu_type_) {
+      case MENU_PRESET:
+        if (SaberBase::IsOn()) {
+          next_preset_fast();
+        } else {
+          next_preset();
+        }
+        break;
+      case MENU_VOLUME:
+         VolumeUp();
+         break;
+      case TRACK_PLAYER:
+        if (track_player_) {
+          track_player_->Stop();
+          track_player_.Free();
+        }
+        track_num_ += 1;
+        if (track_num_ > num_tracks_ - 1) track_num_ = 0;
+        char playtrack[128];
+        RunCommandAndGetSingleLine("list_current_tracks", nullptr, track_num_, playtrack, sizeof(playtrack));
+  #ifdef ENABLE_AUDIO
+        MountSDCard();
+        EnableAmplifier();
+        track_player_ = GetFreeWavPlayer();
+        if (track_player_) {
+          track_player_->Play(playtrack);
+        } else {
+          STDOUT.println("No available WAV players.");
+        }
+  #else
+        STDOUT.println("Audio disabled.");
+  #endif
+        break;
+  #ifdef FETT263_EDIT_MODE_MENU
+      case MENU_TOP:
+        menu_top_pos_ = (menu_top_pos_ + 1) % SUBMENUS;
+        if (menu_top_pos_ == 0) menu_top_pos_ = 1;
+        switch (menu_top_pos_) {
+          case 1:
+            PlayMenuSound("mstyle.wav");
+            break;
+          case 2:
+            PlayMenuSound("mcolor.wav");
+            break;
+          case 3:
+            PlayMenuSound("mfont.wav");
+            break;
+          case 4:
+            PlayMenuSound("mtrack.wav");
+            break;
+          case 5:
+            PlayMenuSound("msetting.wav");
+            break;
+          case 6:
+            PlayMenuSound("mcopy.wav");
+            break;
+          case 7:
+            PlayMenuSound("mdelete.wav");
+            break;
+          case 8:
+            PlayMenuSound("mexit.wav");
+            break;
+       }
+       break;
+  #if NUM_BLADES > 1
+      case MENU_BLADE_STYLE:
+        if (blade_num_ == NUM_BLADES) blade_num_ = 0;
```

Comment:
This is confusing.
I think it should be:

  blade_num_ += 1;
  if (blade_num_ > NUM_BLADES) blade_num_ = 1;

Reply:
updated